### PR TITLE
fix: [CDS-75153]: Add manifest identifier and their location to log when connector ref are wrong

### PR DESCRIPTION
**Pull request by sarthakkasat**
## Describe your changes
Add manifest identifier and their location to log when connector ref are wrong, created a map of manifestId to location and connectorID to set of manifestId to map the values.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51300)

### CODE REVIEW

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java [hunk 1]
```diff
@@ -535,20 +552,31 @@ private void validateConnectors(Ambiance ambiance, List<ManifestAttributes> mani
...
     }
 
-    final Set<String> connectorIdentifierRefs = manifestsToConsider.stream()
-                                                    .map(ManifestAttributes::getStoreConfig)
-                                                    .map(StoreConfig::getConnectorReference)
-                                                    .map(ParameterField::getValue)
-                                                    .collect(Collectors.toSet());
+    Map<String, Set<String>> connectorIdentifierRefs =
+        manifestsToConsider.stream().collect(Collectors.toMap(manifestConfig
+            -> manifestConfig.getStoreConfig().getConnectorReference().getValue(),
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java:568]: manifestConfig.getStoreConfig() can be null

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java [hunk 2]
```diff
@@ -535,20 +552,31 @@ private void validateConnectors(Ambiance ambiance, List<ManifestAttributes> mani
...
+            manifestConfig
+            -> new HashSet<>(Collections.singletonList(manifestConfig.getIdentifier())),
+            (existingSet, newSet) -> {
+              existingSet.addAll(newSet);
+              return existingSet;
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java:573]: can we look to simplify this logic or maybe extract out as a method ?

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java [hunk 3]
```diff
@@ -562,14 +590,39 @@ private void validateConnectors(Ambiance ambiance, List<ManifestAttributes> mani
...
+      List<String> fileLocationList = new ArrayList<>();
+      List<String> manifestIdentifierList = new ArrayList<>();
+      for (String connectorIdentifier : connectorsNotFound) {
+        manifestIdentifierList.add("[" + String.join(",", connectorIdentifierRefs.get(connectorIdentifier)) + "]");
+        fileLocationList.add("["
+            + connectorIdentifierRefs.get(connectorIdentifier)
+                  .stream()
+                  .map(manifestFileLocation::get)
+                  .collect(Collectors.joining(","))
+            + "]");
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java:620]: Lets discuss offline, ideally this logic should be simplified or at least re-used

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

**Review by sarthakkasat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51300)* status: Some of the checks are too old. 1 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java` line 396 at r1](https://reviewable.io/reviews/harness/harness-core/51300#gh-1285396552:-NbIh7Zw67XyRqCL97cn:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ea23c024d61a6f44d5f8bf36e8e7ba60860e8cdf/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java#L396)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

instead of modifying the arguments, lets create a new class
```java
class ManifestsData {
    List<ManifestConfigWrapper> manifests;
    Map<String, String> manifestFileLocation;
}
```
and return that in this method
</blockquote></details>

Done.
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java` line 562 at r1](https://reviewable.io/reviews/harness/harness-core/51300#gh-1285398356:-NbIh8Fo-tfcWsNg3ULD:banmm4c) ([raw file](https://github.com/harness/harness-core/blob/ea23c024d61a6f44d5f8bf36e8e7ba60860e8cdf/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java#L562)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

```text
I think "Connector ref field not present" is misleading. We want to say that the connector does not exist sort of
```
</blockquote></details>

This message is suggesting that connector ref was not provided at all, not that the connector provided does not exist
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java` line 568 at r1](https://reviewable.io/reviews/harness/harness-core/51300#gh-1285418468:-NbIhJjO0k9mLn-asHzG:bimrs1b) ([raw file](https://github.com/harness/harness-core/blob/ea23c024d61a6f44d5f8bf36e8e7ba60860e8cdf/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java#L568)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

manifestConfig.getStoreConfig() can be null 
</blockquote></details>

We are throwing exception before validating connectors if store config is null
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java` line 573 at r1](https://reviewable.io/reviews/harness/harness-core/51300#gh-1285427456:-NbIhUWr9wG-NhLZW3ib:b-fefi7q) ([raw file](https://github.com/harness/harness-core/blob/ea23c024d61a6f44d5f8bf36e8e7ba60860e8cdf/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java#L573)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

can we look to simplify this logic or maybe extract out as a method ?
</blockquote></details>

Extracted as method

**Pull request closed without merging**